src/sssom/io.py

```diff
@@ -130,6 +130,23 @@ def split_file(input_path: str, output_directory: Union[str, Path]) -> None:
     write_tables(splitted, output_directory)
 
 
+def convert_plain_prefix_map_to_extended(prefix_map):
```

Comment:
## `convert_plain_prefix_map_to_extended()`
<details><summary>Details</summary>
<p>

### About
Converts a plain prefix_map to an extended prefix map (EPM).

### Rationale
Why did I make this? I can't say 100% for certain because I'm new to a lot of this code, but from what I can see, there are parts of the code which seem to expect prefix maps in the plain (Dict of PREFIX -> URI_PREFIX) format. And I can't on my own (and don't want to) redesign all of `sssom-py` to use EPMs instead. Therefore I have created this for flexibility, to handle situations where I must deal with plain prefix maps some of the time, but I need functionality of the extended prefix maps (in my case, prefix synonyms).

### Other changes to consider
I wanted to add some typing to this, like one of the following...
```py
def convert_plain_prefix_map_to_extended(prefix_map: Dict[PREFIX, URI]) -> List[Dict]:

def convert_plain_prefix_map_to_extended(prefix_map: PREFIX_MAP) -> EXTENDED_PREFIX_MAP:
```
...but I didn't want to be bogged down with deciding/discussing this, as it's important to merge this soon to fix a bug that's preventing the Mondo pipeline from being run.

</p>
</details>

Reply:
I'm not sure these code changes appropriately address the issue where you have a prefix map that isn't bijective. Please add some unit tests that reproduce the issue you're having then we can take a second pass at the implementation

Reply:
I should use that term, bijective prefix map, if I can figure out exactly one that means. Non-duplicative entries I guess.

I know it addresses my issues because I made the changes locally and they work. But yeah I can add a test.

Reply:
The issue here is this function is non-deterministic in how it picks which prefix becomes the "canonical" one, based on the order of iteration through the prefix map. Some alternatives:

1. Preferred: enforce the MONDO prefix map is bijective
2. Help me push through changes to the SSSOM spec to allow the `curie_map` field to contain an EPM, then we can more carefully define the prefix map
3. Least preferred: add an additional argument to this `convert_plain_prefix_map_to_extended` function that is the list of preferred prefixes, which you have to specify ahead of time, therefore making this function deterministic

Reply:
1. _Bijective enforcement_: An alternative would be something like a `strict` param for bijectiveness, and if `strict=False`, we can do this behavior to solve this problem via inferred prefix synonymity. Else, throw an error.
2. _Spec_: I probably don't have much time to work on the spec. At least for this PR I think @matentzn will probably agree the bugfix should come before official spec change. I would suggest though that rather than (a), `curie_map` allowing for both plain and extended prefix maps, (b) SSSOM should introduce a new field `extended_prefix_map`.
3. _Determinism_: For making this function deterministic, there are other possibilities. Here's another idea, which I prefer: For all cases where the same `uri_prefix` has more than 1 `prefix`, the prefix that comes first alphabetically is inferred as canonical and the others are synonyms. We can also show a warning message when this happens. I think this goes well with my suggestion in (1) for a `strict` param. I prefer this is type of way of handling this because I think we should solve this problem primarily by encouraging extended prefix maps over plain prefix maps whenever possible, but for those who wish to use plain prefix maps, give them the least amount up front work to do; make it an easy user experience.

Reply:
Thank you @joeflack4 for attacking this problem! It is very important you make issues before making PRs, as there are always dozens of unspoken (undocumented) assumptions about the way we designed things - for example in sssom-py. 

Can your problem be solved by making sure that the prefix maps passed to Mondo are bijective? What exactly is the problem in the Mondo pipeline, i.e. why are we even passing non-bijective prefixmaps into the system? Only if we can establish that there is an undeniable use case for non-bijective prefixmaps should we consider interpreting those as EPMs (I am not generally opposed to that, this PR may well have the solution). 

To make an informed call, which may get me to reconsider this PR here, I need to know exactly why the problem occurred in the first place.

If we decide that the use case is compelling, and your solution is the way to go, then we can create a test (an explicit one, see `test_parsers.py` towards the end that illustrates the compelling issue and its solution.

Reply:
Even though I've provided a utility function in `curies` to do this, I still think SSSOM-py should require bijective prefix maps (or if not, require them to be formulated as EPMs)

Reply:
I agree with @cthoyt, for different reasons than him, but the outcome is the same - I think non-bijective prefix maps are, in 85% of the cases, unintentional and should be fixed by the provider rather than processed as EPMs. I fear that interpreting prefixmaps as EPMs in sssom-py will lead to unintended, like cases where the user simply made a typo which is now leading to a wrong reconciliation.

Reply:
Sounds good; I suggested some ways of doing this earlier--passing EPMs instead. The choices:
a. Work w/ the EPMs in `mondo-ingest`; don't need to call `sssom-py` as Harshad initially set up
b. Keep Harhad's original lexmatch scripting pattern, and add a new function to `sssom-py`: `_get_converter_and_metadata_from_epm()`
c. keep Harhad's original lexmatch scripting pattern, and update `_get_converter_and_metadata()` for a new param for EPMs

Since this is now a private function I'm guessing I should try for 'a'.

Reply:
I dont know the concrete thoughts you are having about (a), but I think this should be the right place for the solution.

Reply:
This functionality has now been removed. It is not necessary to solve any problems in https://github.com/monarch-initiative/mondo-ingest/pull/394. It can be re-handled in some variation in future PR for EPM support.